src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -256,26 +253,48 @@ public void reset() {
         String d = serverContext.getDataStore().getLogDir();
         localLog.close();
         if (d != null) {
-            Path dir = FileSystems.getDefault().getPath(d);
-            String prefixes[] = new String[]{"log"};
-
-            for (String pfx : prefixes) {
-                try (DirectoryStream<Path> stream =
-                             Files.newDirectoryStream(dir, pfx + "*")) {
-                    for (Path entry : stream) {
-                        // System.out.println("Deleting " + entry);
-                        Files.delete(entry);
-                    }
-                } catch (IOException e) {
-                    log.error("reset: error deleting prefix " + pfx + ": " + e.toString());
+            Path dir1 = FileSystems.getDefault().getPath(d + "/log");
+            String report = "";
+
+            // Today I learned that the globbing done by newDirectoryStream does not
+            // handle any subdirectory "*/*" matching.  :-(
+            // Use try-with-resources to avoid leaking file descriptor on the directory.
+            try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir1, "*-*-*-*-*")) {
+                for (Path entry : stream) {
+                    report = entry.toString();
+                    log.trace("Deleting recursively " + report);
+                    delete_all_matching_files(entry, "*");
+                    Files.delete(entry);
                 }
+            } catch(IOException e){
+                log.error("reset: error deleting prefix " + report + ": " + e.toString());
+            }
+            try {
+                delete_all_matching_files(dir1, "*");
+            } catch(IOException e){
+                log.error("reset: error deleting prefix " + dir1 + "/*: " + e.toString());
             }
         }
         reboot();
     }
 
+    private void delete_all_matching_files(Path dir, String glob) throws IOException {
+        // Use try-with-resources to avoid leaking file descriptor on the directory.
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(dir, glob)) {
+            for (Path entry : stream) {
+                log.trace("Deleting " + entry);
+                Files.delete(entry);
+            }
+        }
+    }
+
     @Override
     public void reboot() {
```

Comment:
What is the point of this reboot function? It is a lot of extraneous code that needs to be maintained for correctness, which as far as I can see, adds no additional functionality for anything but tests (and is highly dubious for an integration test anyway, because it is a special path.....)

Can you explain why `reboot` is preferable to calling `shutdown()` and then creating a new LogUnit with the same params? (Or from an integration test, why it is preferable to killing the JVM and then restarting it?)